src/codegen/compile.ml

```diff
@@ -4739,7 +4603,7 @@ module Lifecycle = struct
     | InPreUpgrade -> [Idle]
     | PostPreUpgrade -> [InPreUpgrade]
     | InPostUpgrade -> [InInit]
-    | InComposite -> [Idle; InComposite]
+    | InComposite -> [Idle]
```

Comment:
I remove the memory reserve for queries and upgrades, as I think/thought 64-bit space would be sufficient. This pre-state was introduced during that implementation.

There, we used a trick to grant the reserve to the composite callbacks:
```
| Type.Shared Type.Composite ->
    (* Stay in composite query state such that callbacks of 
        composite queries can also use the memory reserve. 
        The state is isolated since memory changes of queries 
        are rolled back by the IC runtime system. *)
        Lifecycle.trans env Lifecycle.InComposite
```